benchmark-java/src/main/java/akka/grpc/benchmarks/driver/LoadClient.java

```diff
@@ -0,0 +1,344 @@
+  private ExecutorService fixedThreadPool;
+  private Messages.SimpleRequest simpleRequest;
+  private final OperatingSystemMXBean osBean;
+  private long lastMarkCpuTime;
+
+  LoadClient(Control.ClientConfig config) throws Exception {
+    log.log(Level.INFO, "Client Config \n" + config.toString());
+    this.config = config;
+
+    this.system = ActorSystem.create("AsyncClient");
+    this.mat = ActorMaterializer.create(system);
+
+    // Create the clients
+    clients = new BenchmarkServiceClient[config.getClientChannels()];
+    for (int i = 0; i < config.getClientChannels(); i++) {
+      InetSocketAddress socketAddress = (InetSocketAddress) Utils.parseSocketAddress(
+          config.getServerTargets(i % config.getServerTargetsCount()));
+
+      GrpcClientSettings settings = Utils.createGrpcClientSettings(socketAddress);
+
+      BenchmarkServiceClient client = BenchmarkServiceClient.create(settings, mat, system.dispatcher());
+
+      clients[i] = client;
+    }
+
+    // Determine no of threads
+    threadCount = config.getAsyncClientThreads() == 0
+        ? Runtime.getRuntime().availableProcessors()
+        : config.getAsyncClientThreads();
+
+    // Use a fixed sized pool of daemon threads.
+    fixedThreadPool = Executors.newFixedThreadPool(threadCount,
+        new DefaultThreadFactory("client-worker", true));
+
+    system.getWhenTerminated().whenComplete((__, t) -> shutdownNow());
+
+    // Create the load distribution
+    switch (config.getLoadParams().getLoadCase()) {
+      case CLOSED_LOOP:
+        distribution = null;
+        break;
+      case LOAD_NOT_SET:
+        distribution = null;
+        break;
+      case POISSON:
+        // Mean of exp distribution per thread is <no threads> / <offered load per second>
+        distribution = new ExponentialDistribution(
+            threadCount / config.getLoadParams().getPoisson().getOfferedLoad());
+        break;
+      default:
+        throw new IllegalArgumentException("Scenario not implemented");
+    }
+
+    // Create payloads
+    switch (config.getPayloadConfig().getPayloadCase()) {
+      case SIMPLE_PARAMS: {
+        Payloads.SimpleProtoParams simpleParams = config.getPayloadConfig().getSimpleParams();
+        simpleRequest = Utils.makeRequest(Messages.PayloadType.COMPRESSABLE,
+            simpleParams.getReqSize(), simpleParams.getRespSize());
+        break;
+      }
+      default: {
+        // Not implemented yet
+        throw new IllegalArgumentException("Scenario not implemented");
+      }
+    }
+
+    List<OperatingSystemMXBean> beans =
+        ManagementFactory.getPlatformMXBeans(OperatingSystemMXBean.class);
+    if (!beans.isEmpty()) {
+      osBean = beans.get(0);
+    } else {
+      osBean = null;
+    }
+
+    // Create the histogram recorder
+    recorder = new Recorder((long) config.getHistogramParams().getMaxPossible(), 3);
+  }
+
+  /**
+   * Start the load scenario.
+   */
+  void start() {
+    Runnable r;
```

Comment:
Why is `r` defined outside of the loop? Could be `final Runnable r;` inside of it